### ISSUE TITLE
JSON Query to graphql/query: 429 Too Many Requests each time when I run the script

### ISSUE THREAD

**Issue by taga93** [instaloader/instaloader#902]
I'm using Instaloader for more than a year, I haven't used Instaloader for more than a week but today when I ran the I got an error:

```
JSON Query to graphql/query: 429 Too Many Requests: redirected to login
```

Then I have tried from my other computer and I got the same error right away (I have tried with on windows and mac). Then my friend tried from his house/his machine and he also get the same error right away.
This is the full error:

```
HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/
JSON Query to graphql/query: 429 Too Many Requests: redirected to login [retrying; skip with ^C]
Requests within last 10/11/20/22/30/60 minutes grouped by type:
                              other:    0    0    1    1    1    1
   2b0673e0dc4580674a88d426fe00ea90:    0    0    1    1    1    1
 * 97b41c52301f77ce508f55e66d17620e:    1    1    2    2    2    2
Instagram responded with HTTP error "429 - Too Many Requests". Please
do not run multiple instances of Instaloader in parallel or within
short sequence. Also, do not use any Instagram App while Instaloader
is running.
The request will be retried in 666 seconds, at 17:44.
```

Im using the latest version on Instaloader.

I know that this error pops up when max request number is reached, but now it happens right away and thats strange to me.
It happens after 1 or 2 requests

**Comment by aandergr** [comment]
Thanks a lot for reporting this! [Version 4.5.5](https://github.com/instaloader/instaloader/releases/tag/v4.5.5) has this fixed.

When requested data is restricted for anonymous access, Instagram responds with a HTTP redirect to the login page. Instaloader handled this kind of response erroneously as a "Too Many Requests" response instead of what it actually is, a notification that login is required to access that data.

**Comment by taga93** [comment]
My version is Instaloader 4.5.4. When I want to upgrade with pip install --upgrade Instaloader it says that I have latest version.
I saw that on PyPl it says that 4.5.4. is latest version.
How can I upgrade it?

In my code I put time.sleep(5) before analyzing each post. But after 10 posts I get the same error.

And my instagram account got blocked.

**Comment by aandergr** [comment]
Version 4.5.5 is on PyPI since today, 14:41 UTC. You can upgrade with

```
pip3 install --upgrade instaloader
```

**Comment by taga93** [comment]
I have updated the lib, now its the latest version, and again

Too many queries in the last time. Need to wait 64 seconds, until 17:54.
Too many queries in the last time. Need to wait 14 seconds, until 17:56.
...
and then it breaks
and this keeps showing every time.
And I have time.sleep(5) , and I have logged in to ig account.

user_agent = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/601.7.8 (KHTML, like Gecko)'
L = instaloader.Instaloader(max_connection_attempts = 3, user_agent = user_agent)
L.login('xxx', 'xxx') 
profile_posts = instaloader.Profile.from_username(L.context, 'xxx').get_posts()

for post in profile_posts :
    for com in post..get_comments():
        print(com)

I'm trying to fetch comments from each post (each post has between 10 and 200 comments)
